### ISSUE TITLE
SC2310 don't report if || exit or || return is used

### ISSUE THREAD

**Issue by robstoll** [koalaman/shellcheck#2532]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash
	
# we want to be sure that we return at this point even if someone has `set +e` thus the or clause
checkParameterDefinitionIsTriple paramArr1 || return $?
```

#### Here's what shellcheck currently says:
```
In /home/tegonal/projects/ght/scripts/scripts/../src/utility/parse-args.sh line 160:
        checkParameterDefinitionIsTriple paramArr1 || return $?
        ^-- SC2310 (info): This function is invoked in an || condition so set -e will be disabled. Invoke separately if failures should cause the script to exit.
```
#### Here's what I wanted or expected to see:
no error, as I do `|| return $?`

**Comment by koalaman** [comment]
It's referring to failures within the function, which no longer cause an exit due to this invocation. For example:

```
set -e
foo() {
  scp file backup:archive/
  rm file
}

# Works: script stops if scp fails
foo
# Fails: script deletes the file if scp fails
foo || exit $?
```

It's possible (and given the name, very likely) that `checkParameterDefinitionIsTriple` does not depend on `set -e` being enabled to function correctly and that this code is therefore fine, but ShellCheck is currently not able to determine this.

**Comment by robstoll** [comment]
Sorry, I realised my false assumption but forgot to close this issue. Thanks for the additional explanation.
